src/oci.c

```diff
@@ -261,35 +267,31 @@ cc_oci_kill (struct cc_oci_config *config,
 		goto error;
 	}
 
-	/* send signal to cc-shim because it
-	 * will send killcontainer to hyperstart
-	 */
-	if (kill (state->pid, signum) < 0) {
-		g_critical ("failed to stop container %s "
-				"running with pid %u: %s",
-				config->optarg_container_id,
-				(unsigned)state->pid,
-				strerror (errno));
+#ifndef UNIT_TESTING
+	if (! cc_proxy_hyper_kill_container (config, signum, allProcesses)) {
+		g_critical ("failed to kill container %s: %s",
+			config->optarg_container_id,
+			strerror (errno));
 		/* revert container status */
 		config->state.status = last_status;
 		if (! cc_oci_state_file_create (config, state->create_time)) {
 			g_critical ("failed to recreate state file");
 		}
 		return false;
 	}
+#endif //UNIT_TESTING
 
-#ifndef UNIT_TESTING
-	/* cc-shim is not able to catch SIGKILL and SIGSTOP
-	 * signals for this reason we must kill the container
-	 * using hyperstart's API
-	 */
-	if (signum == SIGKILL || signum == SIGSTOP) {
-		if (! cc_proxy_hyper_kill_container (config, signum)) {
-			g_critical ("failed to kill container");
+	if (signum == SIGKILL) {
+		/* send signal to cc-shim because SIGKILL will end
+		 * the VM and cc-shim is not able to catch it
+		 */
+		if (kill (state->pid, signum) < 0) {
+			g_critical ("failed to stop cc-shim %u: %s",
+					(unsigned)state->pid,
+					strerror (errno));
```

Comment:
I kill container's processes before cc-shim because cc-shim will catch those signals (exit codes) and will end.
cc-shim is not able to catch SIGKILL hence I have to kill it too
Adding comments ..
Thank you!